text/0000-add-logical-operators.md

```diff
@@ -0,0 +1,95 @@
+- Start Date: 2019-12-08
+- Relevant Team(s): (fill this in with the [team(s)](README.md#relevant-teams) to which this RFC applies)
+- RFC PR: https://github.com/emberjs/rfcs/pull/562
+- Tracking: (leave this empty)
+
+# Adding Logical Operators to Templates
+
+## Summary
+
+Add new built-in template `{{and}}`, `{{or}}` and `{{not}}` helpers to perform basic logical operations in templates, similar to those included in `ember-truth-helpers`.
+
+This RFC is a subset of the changes proposed in #388.
+
+## Motivation
+
+It is a very common need in any sufficiently complex Ember app to perform some logical comparison operations and often the most convenient place to do it is right in the templates.
+Because of that, [ember-truth-helpers](https://github.com/jmurphyau/ember-truth-helpers) is one of the most installed addons out there, either directly by apps or indirectly by
+other addons that those apps consume.
+
+The fact that `ember-truth-helpers` is so popular is a good signal that this it is filling a perceived gap in Ember's functionality.
+
+A second reason is that it might help make Ember more approachable to newcomers that have some experience in other frameworks.
+Most if not all web frameworks have some way of comparing numbers in the templates and it's surprising that Ember requires an third party package to perform
+even the most basic operations.
+
+A third reason, this time technical, is that by implementing those helpers at a lower-level, we can make them more performant by making them short-circuit.
+Right now helpers implemented using public APIs like those in `ember-truth-helpers` eagerly consume their arguments. In the case of logical operations like `{{and a b}}`,
+once the first argument (`a`) is evalued to _falsey_, the second argument (`b`) is irrelevant. Sometimes arguments can be expensive properties to calculate,
+and by short-circuiting we can avoid computing them at all sometimes.
+
+
+## Detailed design
+
+Add `{{and}}`, `{{or}}` and `{{not}}` helpers.
+
+#### `{{and}}`
+Takes at least two positional arguments, bound or unbound. Raises an error if invoked with less than two arguments.
```

Comment:
For those of us following along who are less intimately acquainted with some of these details 😅 can you explain (not necessarily in this paragraph!) what is meant by “bound or unbound”? Thanks!

Reply:
I meant that you can pass bound arguments like local properties/arguments (e.g `{{or this.someProp @someArg}}`) or literal values like strings/numbers/booleans (e.g. `{{or @someArg "defaultValue"}}`)

I don't know if there is a better term for that.

Reply:
Ah, got it. I’d probably drop it entirely as unnecessary, then, unless there’s a specific reason to call it out. 🤔

Reply:
You're maybe right, all helpers allow both afaik. I'll drop it.